// material-ui/List/ListItemAvatar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.styleSheet = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

exports.default = ListItemAvatar;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _warning = require('warning');

var _warning2 = _interopRequireDefault(_warning);

var _jssThemeReactor = require('jss-theme-reactor');

var _customPropTypes = require('../utils/customPropTypes');

var _customPropTypes2 = _interopRequireDefault(_customPropTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styleSheet = exports.styleSheet = (0, _jssThemeReactor.createStyleSheet)('MuiListItemAvatar', function () {
  return {
    dense: {
      height: 32,
      marginRight: 8,
      width: 32
    }
  };
}); //  weak

function ListItemAvatar(props, context) {
  if (!context.dense) {
    process.env.NODE_ENV !== "production" ? (0, _warning2.default)(false, 'Material-UI: <ListItemAvatar/> is a simple wrapper for the dense mode.\n      You do not need it.') : void 0;
    return props.children;
  }

  var children = props.children,
      classNameProp = props.className,
      other = (0, _objectWithoutProperties3.default)(props, ['children', 'className']);

  var classes = context.styleManager.render(styleSheet);

  return _react2.default.cloneElement(children, (0, _extends3.default)({
    className: (0, _classnames2.default)(classes.dense, classNameProp, children.props.className)
  }, other));
}

ListItemAvatar.propTypes = {
  children: _react.PropTypes.element.isRequired,
  /**
   * The CSS class name of the root element.
   */
  className: _react.PropTypes.string
};

ListItemAvatar.contextTypes = {
  dense: _react.PropTypes.bool,
  styleManager: _customPropTypes2.default.muiRequired
};
